CI random failure: invalid memory address or nil pointer dereference: leveldb.(*DB).isClosed(...)

https://drone.gitea.io/go-gitea/gitea/57643/2/12

(Hmm ... duplicated with https://github.com/go-gitea/gitea/issues/19957)

```
2022/07/15 07:13:08 ...ueue_channel_test.go:170:func3() [I] Finally terminating
2022/07/15 07:13:08 ...eue/queue_channel.go:154:func1() [W] ChannelQueue:  Terminated before completed flushing
2022/07/15 07:13:08 ...eue/queue_channel.go:154:func1() [W] ChannelQueue: first-channel Terminated before completed flushing
2022/07/15 07:13:08 ...eue/queue_channel.go:154:func1() [W] ChannelQueue: second-channel Terminated before completed flushing
2022/07/15 07:13:08 ...ueue_disk_channel.go:185:func3() [W] [62d11384-3] LevelQueue: second-level shut down before completely flushed
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xc893a1]

goroutine 136 [running]:
github.com/syndtr/goleveldb/leveldb.(*DB).isClosed(...)
	/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_state.go:230
github.com/syndtr/goleveldb/leveldb.(*DB).ok(...)
	/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db_state.go:235
github.com/syndtr/goleveldb/leveldb.(*DB).Get(0x0, {0xc000143a98, 0x8, 0x8}, 0xc0003221e8?)
	/go/pkg/mod/github.com/syndtr/goleveldb@v1.0.0/leveldb/db.go:838 +0x81
gitea.com/lunny/levelqueue.(*Queue).RPop(0xc0001f4080)
	/go/pkg/mod/gitea.com/lunny/levelqueue@v0.4.1/queue.go:198 +0x327
code.gitea.io/gitea/modules/queue.(*LevelQueueByteFIFO).Pop(0xc0001b6168, {0x0?, 0x3?})
	/drone/src/modules/queue/queue_disk.go:104 +0x52
code.gitea.io/gitea/modules/queue.(*ByteFIFOQueue).doPop(0xc0003121e0)
	/drone/src/modules/queue/queue_bytefifo.go:299 +0x128
code.gitea.io/gitea/modules/queue.(*ByteFIFOQueue).readToChan(0xc0003121e0)
	/drone/src/modules/queue/queue_bytefifo.go:234 +0x7e5
code.gitea.io/gitea/modules/queue.(*ByteFIFOQueue).Run(0xc0003121e0, 0x10e12a8, 0x10e12b0)
	/drone/src/modules/queue/queue_bytefifo.go:164 +0x413
created by code.gitea.io/gitea/modules/queue.(*PersistableChannelQueue).Run
	/drone/src/modules/queue/queue_disk_channel.go:178 +0x67d
FAIL	code.gitea.io/gitea/modules/queue	0.556s
```

--- a/go.mod
+++ b/go.mod
@@ -9,7 +9,7 @@ require (
 	gitea.com/go-chi/cache v0.2.0
 	gitea.com/go-chi/captcha v0.0.0-20211013065431-70641c1a35d5
 	gitea.com/go-chi/session v0.0.0-20211218221615-e3605d8b28b8
-	gitea.com/lunny/levelqueue v0.4.1
+	gitea.com/lunny/levelqueue v0.4.2-0.20220729054728-f020868cc2f7
 	github.com/42wim/sshsig v0.0.0-20211121163825-841cf5bbc121
 	github.com/NYTimes/gziphandler v1.1.1
 	github.com/PuerkitoBio/goquery v1.8.0

